src/go/wsl-helper/pkg/wsl-utils/dism_windows.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright © 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wslutils
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+type dismSession uint
+
+const (
+	// kWindowsFeature is the name of the Windows feature that needs to be
+	// installed.
+	kWindowsFeature = "VirtualMachinePlatform"
+	// DISM_ONLINE_IMAGE is the DISM "image path" that signifies we're trying
+	// to modify the running Windows installation.
+	DISM_ONLINE_IMAGE         = "DISM_{53BFAE52-B167-4E2F-A258-0A37B57FF845}"
+	DismLogErrorsWarningsInfo = 2
+)
+
+var (
+	dllDismApi        = windows.NewLazySystemDLL("dismapi.dll")
+	dismInitialize    = dllDismApi.NewProc("DismInitialize")
+	dismOpenSession   = dllDismApi.NewProc("DismOpenSession")
+	dismCloseSession  = dllDismApi.NewProc("DismCloseSession")
+	dismEnableFeature = dllDismApi.NewProc("DismEnableFeature")
+)
+
+func errorFromHResult(hr int32, err error) error {
+	var result error
+	if hr < 0 {
+		result = windows.Errno(hr)
+		if err != nil {
+			result = fmt.Errorf("%w: %w", result, err)
+		}
+		return result
+	}
+	return nil
+}
+
+// DismDoInstall installs the Virtual Machine Platform Windows feature.
+func DismDoInstall(ctx context.Context, log *logrus.Entry) error {
+	var session dismSession
+
+	hr, _, err := dismInitialize.Call(
+		uintptr(DismLogErrorsWarningsInfo),
+		uintptr(unsafe.Pointer(nil)),
+		uintptr(unsafe.Pointer(nil)),
+	)
+	err = errorFromHResult(int32(hr), err)
+	if err != nil {
+		return fmt.Errorf("error initializing DISM: %w", err)
+	}
+
+	buf, err := windows.UTF16PtrFromString(DISM_ONLINE_IMAGE)
+	if err != nil {
+		log.WithError(err).Error("Failed to convert DISM_ONLINE_IMAGE")
+		return err
+	}
+	rv, _, err := dismOpenSession.Call(
```

Comment:
Shouldn't the `rv` be called `hr`? From https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism/dismopensession-function?view=windows-11 I think OpenSession returns a `* DimSession` and if I understand the go/C bridge correct here, you're ignoring that result (I assume the `err` variable is for an error in the go/C bridge, and `rv` is the `HRESULT`